pkg/plugin/pr-sanitizer/event_handler_unit_test.go

```diff
@@ -33,45 +30,67 @@ var _ = Describe("PR-Sanitizer Plugin features", func() {
 
 		DescribeTable("should not recognize PR as compliant with semantic commit message if title doesn't start with any default prefix",
 			func(title string) {
-				Expect(handler.HasTitleWithValidType(prsanitizer.PluginConfiguration{}, title)).To(BeFalse())
+				prefixes := prsanitizer.GetValidTitlePrefixes(prsanitizer.PluginConfiguration{})
+				Expect(prsanitizer.HasTitleWithValidType(prefixes, title)).To(BeFalse())
 			},
 			Entry("not a supported semantic commit prefix", "wip-fix off-by one bug"),
+			Entry("title starting by fixes", "fixes failing test"),
 			Entry("empty title", ""),
 			Entry("nil title", nil),
 		)
 	})
 
 	Context("Description verifier", func() {
 
-		var handler *prsanitizer.GitHubPRSanitizerEventsHandler
+		DescribeTable("should recognize issue link presence",
+			func(desc string) {
+				pr := &github.PullRequest{Body: utils.String(desc)}
+				msg := prsanitizer.CheckIssueLinkPresence(pr, prsanitizer.PluginConfiguration{}, log.NewTestLogger())
+				Expect(msg).To(BeEmpty())
+			},
+			Entry("fixes keyword with surrounded text", "PR\r\n\r\nfixes: #1 issue"),
+			Entry("closed keyword inline", "PR closed: org/repo#1 issue"),
+			Entry("resolve keyword with other repo path", "PR\r\nresolve: org/my-repo#1"),
+			Entry("only fixed keyword with link", "fixed :  #1"),
+			Entry("closes keyword with other repo path inline", "PR closes org/repo#1 issue"),
+			Entry("two links", "pr resolves org/my-repo#1 and fixes: #1"),
+		)
 
-		BeforeEach(func() {
-			handler = &prsanitizer.GitHubPRSanitizerEventsHandler{Client: NewDefaultGitHubClient()}
-		})
+		DescribeTable("should NOT recognize issue link presence",
+			func(desc string) {
+				pr := &github.PullRequest{Body: utils.String(desc)}
+				msg := prsanitizer.CheckIssueLinkPresence(pr, prsanitizer.PluginConfiguration{}, log.NewTestLogger())
+				Expect(msg).NotTo(BeEmpty())
+			},
+			Entry("missing hash sign", "PR\r\n\r\nfixes: 1 issue"),
```

Comment:
Might be my obsession, but can we make some tests with uppercase keywords too?

Reply:
I like this obsession :-P 6369164